tests/test_pyxis.py

```diff
@@ -0,0 +1,426 @@
+        mock.side_effect = side_effect
+        return new_mock
+
+    @patch('freshmaker.pyxis.HTTPKerberosAuth')
+    @patch('freshmaker.pyxis.requests.get')
+    def test_make_request(self, get, auth):
+        get.return_value = self.response
+        test_params = {'key1': 'val1'}
+        self.px._make_request('test', test_params)
+
+        get_url = self.fake_server_url + 'v1/test'
+        self.response.json.assert_called_once()
+        test_params['page_size'] = "100"
+        get.assert_called_once_with(get_url, params=test_params, auth=auth(),
+                                    timeout=conf.net_timeout)
+
+    @patch('freshmaker.pyxis.HTTPKerberosAuth')
+    @patch('freshmaker.pyxis.requests.get')
+    def test_pagination(self, get, auth):
+        my_get = self.copy_call_args(get)
+        self.response.json.side_effect = [
+            {"page": 0, "data": "fake_data1"},
+            {"page": 1, "data": "fake_data2"},
+            {"page": 2}
+        ]
+        my_get.return_value = self.response
+        test_params = {'include': ['total', 'field1']}
+        entity = 'test'
+        auth.return_value = 1
+        self.px._pagination(entity, test_params)
+
+        url = self.fake_server_url + 'v1/test'
+        self.assertEqual(get.call_count, 3)
+        default_params = {'page_size': '100', 'include': ['total', 'field1']}
+        calls = [call(url, params={**default_params, 'page': 0}, auth=1,
+                      timeout=conf.net_timeout),
+                 call(url, params={**default_params, 'page': 1}, auth=1,
+                      timeout=conf.net_timeout),
+                 call(url, params={**default_params, 'page': 2}, auth=1,
+                      timeout=conf.net_timeout)]
+        my_get.assert_has_calls(calls, any_order=True)
+
+    @patch.object(conf, 'pyxis_index_image_organization', new='org')
+    @patch('freshmaker.pyxis.Pyxis._pagination')
+    def test_get_operator_indices(self, page):
+        self.px.get_operator_indices({'param1': 1})
+        page.assert_called_once_with(
+            'operators/indices', {'param1': 1, 'filter': 'organization==org'})
+
+    @patch('freshmaker.pyxis.Pyxis._pagination')
+    def test_get_bundles_per_index_image(self, page):
+        page_copy = self.copy_call_args(page)
+        page_copy.side_effect = [self.bundles[0:2], self.bundles[2:]]
+        out = self.px._get_bundles_per_index_image(self.indices, {"param1": 1})
+
+        include_fields = ['data.channel_name', 'data.version',
+                          'data.related_images', 'data.bundle_path_digest',
+                          'data.bundle_path']
+        expected_params = {'param1': 1, 'include': ','.join(include_fields)}
+        expected_out = {"path/to/registry:v4.5": self.bundles[0:2],
+                        "path/to/registry:v4.6": self.bundles[2:]}
+
+        self.assertEqual(out, expected_out)
+        call_count = 0
+        for index in self.indices:
+            if not index.get('path'):
```

Comment:
This is essentially reimplementing the logic from `_get_bundles_per_index_image`. This is pretty minor here but in general, you want to avoid that since the logic in the actual code could be flawed and you don't want the tests to be using the same logic and not catch that the logic was flawed.

My suggestion is to just to do `assert_has_calls` on the mock object for `_pagination`. That makes the test clearer and less error prone.